contracts/AddressSet.sol

```diff
@@ -15,11 +15,17 @@ library AddressSet {
     }
 
     function remove(Data storage self, address element) {
+        // preserves initial order of elements, can be optimized if not required
+
         uint256 position = self.index[element];
         if (position != 0) {
-            self.list[position - 1] = self.list[self.list.length - 1];
-            self.list.length--;
+            // position starts from 1
+            while (position < self.list.length) {
+                self.list[position - 1] = self.list[position];
+                position++;
+            }
 
+            self.list.length--;
```

Comment:
So, this potentially generates a lot of state changes and might cost lots of gas. Have you considered eg. storing an `mapping (address => boolean) revoked` to keep track of revoked signatures?

Reply:
I think we should postpone performance considerations until we analyze the use cases.